distribution/lib/Standard/Base/0.0.0-dev/src/Internal/Data_Read_Helpers.enso

```diff
@@ -28,8 +31,18 @@ looks_like_uri path:Text -> Boolean =
    A common implementation for fetching a resource and decoding it,
    following encountered data links.
 fetch_following_data_links (uri:URI) (method:HTTP_Method = HTTP_Method.Get) (headers:Vector = []) format =
-    response = HTTP.fetch uri method headers
-    decode_http_response_following_data_links response format
+    fetch_and_decode =
+        response = HTTP.fetch uri method headers
+        decode_http_response_following_data_links response format
+
+    error_handler attempt =
+        caught_error ->
+            if method != HTTP_Method.Get || attempt > 2 then Error.throw caught_error else
+                sleep_time = [100, 200, 400] . get attempt
```

Comment:
Loose suggestion:
```suggestion
			sleep_intervals = [100, 200, 400]
            if method != HTTP_Method.Get || attempt >= sleep_intervals.length then Error.throw caught_error else
                sleep_time = sleep_intervals.get attempt
```

This way the code binds the retry count with the sleep intervals array - so we don't risk an index-out-of-bounds error when tweaking the retry count and forgetting to also update the array.